lib/puppet/type/cs_colocation.rb

```diff
@@ -73,20 +74,53 @@ def should=(value)
   end
 
   autorequire(:cs_shadow) do
-    [ @parameters[:cib] ]
+    autos = []
+    if @parameters[:cib]
+      autos << @parameters[:cib].value
+    end
+
+    autos
+  end
+
+  if Puppet::PUPPETVERSION.to_f >= 4.0
```

Comment:
The version comparison based on `to_f`. Sorry for being unspecific.

I realize that this is in wide use, but it is terrible, because `'3.10'.to_f < '3.8'.to_f`.